planner/core/task.go

```diff
@@ -1161,6 +1161,14 @@ func (p *PhysicalUnionAll) attach2MppTasks(tasks ...task) task {
 }
 
 func (p *PhysicalUnionAll) attach2Task(tasks ...task) task {
+	for _, t := range tasks {
+		if _, ok := t.(*mppTask); ok && p.TP() == plancodec.TypePartitionUnion {
+			// In attach2MppTasks(), will attach Union to mppTask directly.
+			// But PartitionUnion cannot pushdown to tiflash, so disable PartitionUnion pushdown to tiflash explicitly.
+			// TODO: Let mppTask convertToRootTask instead of return invalidTask directly.
```

Comment:
I'm a little confused about the change and the TODO here, if the goal is to `Let mppTask convertToRootTask instead of return invalidTask directly`, seems just delete the following code is enough?
https://github.com/pingcap/tidb/blob/fd5e67581b5a43a1a7d2cbdb8c345678502f9f03/planner/core/task.go#L1164-L1168

Reply:
The goal is to explicitly disable `PartitionUnion` attach to mppTask.
But this code(return invalidTask immediately) will ignore letting childTask to convert to rootTask, then attach PartitionUnoin to converted rootTask. So i added this todo.

Reply:
Comments refined.